src/tcpstream.c

```diff
@@ -517,26 +520,32 @@ int ntttcp_server_epoll(struct ntttcp_stream_server *ss)
 			}
 			/* handle data from an EXISTING client */
 			else {
-				bzero(buffer, ss->recv_buf_size);
-				bytes_to_be_read = ss->is_sync_thread ? 1 : ss->recv_buf_size;
-
-				/* got error or connection closed by client */
-				if ((nbytes = n_read(current_fd, buffer, bytes_to_be_read)) <= 0) {
-					if (nbytes == 0) {
-						ASPRINTF(&log, "socket closed: %d", i);
-						PRINT_DBG_FREE(log);
+				for (int max_io = 0; max_io < MAX_IO_PER_POLL; max_io++) {
```

Comment:
Why we need this?

Reply:
Without this loop upper bound, the benchmark could be monopolized by a single socket that never blocks.

The old code avoided this situation by calling poll() for each iteration of the loop, which was inefficient.